### PR TITLE
Feature #5

### LINKED ISSUE #5: 5) Feature - Implement a read status for messages
We want to store a status for each message reflecting if it has been read by the recipient, and display how many unread messages there are on the front end.
You can reference the spec in [this zip folder](https://storage.googleapis.com/hatchways-app.appspot.com/assessments/data/project-overview/4f81e627-c085-4583-a9ef-bf69316a7a55/assets%202.zip) for how to display the unread messages count. Please include a screenshot in your PR description showing your updated UI.

### PR DESCRIPTION
**Pull request by khzainulabidin**
### Fixes #5 

Both frontend and backend part has been completed. Here is the relevant commit: 77d982f87bb18225c88df30be036787c5c10cebc

Updated UI:

image

Thank you!

### CODE REVIEW

File: server/routes/api/messages.js [hunk 1]
```diff
@@ -45,4 +45,14 @@ router.post("/", async (req, res, next) => {
   }
 });
 
+router.put('/read', async (req, res, next) => {
```
Comment (0916dhkim) [inline server/routes/api/messages.js:48]: Good fix :+1:

File: client/src/utils/socket.js [hunk 2]
```diff
@@ -6,21 +6,50 @@ import {
   addOnlineUser,
 } from "../store/conversations";
 
-const socket = io(window.location.origin);
+export const socketWrapper = {
```
Comment (0916dhkim) [inline client/src/utils/socket.js:9]: What is the difference between doing this and `export const socket = null` ?

**Review by 0916dhkim — APPROVED** [review]
@khzainulabidin Nice work! This PR is now good to be merged :+1: 
There are few things you might want to fix before merging:
- I see you have updated `onlineUsers` from `[]` to `{}`. It is a good change for overall, but it broke the search feature.
- Unread badge only appears after page refresh. It will be useful if the badge shows up real-time.

File: client/src/utils/socket.js (hunk 2 above)
Comment (khzainulabidin) [reply client/src/utils/socket.js:9]: If I initialize it as `const` then I won't be able to assign it a value later on. In other words, I won't be able to initialize it. So I either have to do it like this `export let socket = null` or use the current approach.

**Comment by khzainulabidin** [comment]
Hi @0916dhkim, I have implemented all the changes as requested!

**Pull request merged**